src/pyhf/exceptions/__init__.py

```diff
@@ -175,3 +176,15 @@ def __init__(self, result):
             result, 'message', "Unknown failure. See fit result for more details."
         )
         super().__init__(message)
+
+
+# Deprecated APIs
+def _deprecated_api_warning(
```

Comment:
This could be put somewhere other than `pyhf/exceptions` but I didn't have a good idea in the moment. Welcome to suggestions.